regex/tests/Test.lean

```diff
@@ -153,4 +153,907 @@ def word_class := Regex.parse! r##"\w+"##
+
+-- name = "unicode3-only-ascii"
+#guard unicode_word_boundary_01.captureAll "áxβ" = #[
+  ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+#guard unicode_word_boundary_01.bt.captureAll "áxβ" = #[
+  ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+]
+
+def unicode_non_word_boundary_01 := Regex.parse! r##"\Bx\B"##
+
+-- FIXME: This test is not working as expected.
+-- name = "unicode4"
+-- #eval unicode_word_boundary_02.captureAll "áxβ"
+-- #guard unicode_word_boundary_02.captureAll "áxβ" = #[
+--   ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+-- ]
+-- #guard unicode_word_boundary_02.bt.captureAll "áxβ" = #[
+--   ⟨"áxβ", #[.some ⟨2⟩, .some ⟨3⟩]⟩
+-- ]
+
+-- name = "unicode4-only-ascii"
+#guard unicode_non_word_boundary_01.captureAll "áxβ" = #[]
+#guard unicode_non_word_boundary_01.bt.captureAll "áxβ" = #[]
+
+-- The same as above, but with \b instead of \B as a sanity check.
+def unicode_word_boundary_02 := Regex.parse! r##"\b"##
+
+-- FIXME: This test is not working as expected.
+-- -- name = "unicode5"
+-- #eval unicode_word_boundary_02.captureAll "0\U0007EF5E"
+-- #guard unicode_word_boundary_02.captureAll "0\U0007EF5E" = #[
+--   ⟨"0\U0007EF5E", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+--   ⟨"0\U0007EF5E", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+-- ]
+-- #guard unicode_word_boundary_02.bt.captureAll "0\U0007EF5E" = #[
+--   ⟨"0\U0007EF5E", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+--   ⟨"0\U0007EF5E", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+-- ]
+
+-- FIXME: This test is not working as expected.
+-- -- name = "unicode5-only-ascii"
+-- #eval unicode_word_boundary_02.captureAll "0\U0007EF5E"
+-- #guard unicode_word_boundary_02.captureAll "0\U0007EF5E" = #[
+--   ⟨"0\U0007EF5E", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+--   ⟨"0\U0007EF5E", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+-- ]
+-- #guard unicode_word_boundary_02.bt.captureAll "0\U0007EF5E" = #[
+--   ⟨"0\U0007EF5E", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+--   ⟨"0\U0007EF5E", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+-- ]
+
+-- name = "unicode5-noutf8"
+#guard unicode_word_boundary_02.captureAll "0\xFF\xFF\xFF\xFF" = #[
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+]
+#guard unicode_word_boundary_02.bt.captureAll "0\xFF\xFF\xFF\xFF" = #[
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+]
+
+-- name = "unicode5-noutf8-only-ascii"
+#guard unicode_word_boundary_02.captureAll "0\xFF\xFF\xFF\xFF" = #[
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+]
+#guard unicode_word_boundary_02.bt.captureAll "0\xFF\xFF\xFF\xFF" = #[
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨0⟩, .some ⟨0⟩]⟩,
+  ⟨"0\xFF\xFF\xFF\xFF", #[.some ⟨1⟩, .some ⟨1⟩]⟩
+]
+
+-- Weird special case to ensure that ASCII \B treats each individual code unit
+-- as a non-word byte. (The specific codepoint is irrelevant. It's an arbitrary
+-- codepoint that uses 4 bytes in its UTF-8 encoding and is not a member of the
+-- \w character class.)
+
+def unicode_non_word_boundary_02 := Regex.parse! r##"\B"##
```

Comment:
The entire test cases for `unicode_non_word_boundary_02` are uninteresting to us (since we don't intend to support ASCII mode). Let's delete them.